extensions/wikia/AdEngine/js/lookup/lookupFactory.js

```diff
@@ -4,20 +4,20 @@ define('ext.wikia.adEngine.lookup.lookupFactory', [
 	'ext.wikia.adEngine.adTracker',
 	'ext.wikia.aRecoveryEngine.adBlockDetection',
 	'wikia.lazyqueue',
-	'wikia.log'
-], function (adContext, adTracker, adBlockDetection, lazyQueue, log) {
+	'wikia.log',
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+], function (adContext, adTracker, adBlockDetection, lazyQueue, log, mercuryListener) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has 6 parameters which is greater than 4 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)